Source/WebCore/layout/formattingContexts/inline/InlineContentBalancer.cpp

```diff
@@ -157,14 +158,16 @@ void InlineContentBalancer::initialize()
         previousLineEnd = layoutRange.start;
         previousLine = PreviousLine { lineIndex, lineLayoutResult.contentGeometry.trailingOverflowingContentWidth, !lineLayoutResult.inlineContent.isEmpty() && lineLayoutResult.inlineContent.last().isLineBreak(), !lineLayoutResult.inlineContent.isEmpty(), lineLayoutResult.directionality.inlineBaseDirection, WTFMove(lineLayoutResult.floatContent.suspendedFloats) };
         lineIndex++;
+        if (numberOfVisibleLinesConsideringLineClamp && lineIndex >= numberOfVisibleLinesConsideringLineClamp)
+            break;
```

Comment:
nit: you can bail out a lot earlier.

Reply:
Don't we have to initialize the Balancer until we would reach the last line before clamping?

Reply:
when you say Balancer are you referring to InlineContentBalancer? -just wanna make sure we are talking about the same thing.

Reply:
Yes, exactly!

Reply:
ok, yes we have to initialize the InlineContentBalancer, but we are _inside_ InlineContentBalancer. What I was trying to say with my original comment was please look and make sure we don't compute anything redundant here (i.e. perhaps moving this check a few lines above? -maybe "a lot earlier" was a bit of an exaggeration)